### PR TITLE
factory, adapter, decorator and proxy patterns

### PR DESCRIPTION
**Pull request by mkharkhu**

### CODE REVIEW

File: hw21/decorator/Cheese.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package lesson21.hw21.decorator;
+
+public class Cheese extends Sandwich {
```
Comment (Yanetta) [inline hw21/decorator/Cheese.java]: I think the name of this class could be CheeseSandwich

File: hw21/adapter/ZhigulyAdapter.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package lesson21.hw21.adapter;
+
+public class ZhigulyAdapter implements OldCar {
+
+    NewCar newCar;
```
Comment (SergeyPasko) [inline hw21/adapter/ZhigulyAdapter.java]: private

File: hw21/decorator/Sandwich.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package lesson21.hw21.decorator;
+
+public abstract class Sandwich {
+
+    String desctription = "Slice of bread";
```
Comment (SergeyPasko) [inline hw21/decorator/Sandwich.java]: PRIVATE!!!

File: hw21/proxy/ProxyWebsite.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    private RealWebsite realWebsite;
+    private String siteName;
+
+    public ProxyWebsite(String siteName) {
+        this.siteName = siteName;
+    }
+
+    @Override
+    public void download() {
+        if (realWebsite == null) {
```
Comment (SergeyPasko) [inline hw21/proxy/ProxyWebsite.java]: What this give for you?

**Pull request merged**